nomos-membership/src/membership.rs

```diff
@@ -0,0 +1,509 @@
+use std::collections::HashMap;
+
+use nomos_core::{
+    block::BlockNumber,
+    sdp::{FinalizedBlockEvent, ServiceType},
+};
+
+use crate::{
+    Membership, MembershipConfig, MembershipError, NewSesssion, session::Session,
+    storage::MembershipStorage,
+};
+
+pub struct PersistentMembership<S: MembershipStorage> {
+    storage: S,
+    active_sessions: HashMap<ServiceType, Session>,
+    forming_sessions: HashMap<ServiceType, Session>,
+    latest_block_number: BlockNumber,
+    session_sizes: HashMap<ServiceType, u32>,
+}
+
+#[async_trait::async_trait]
+impl<S> Membership for PersistentMembership<S>
```

Comment:
Could we remove the Membership trait and just have Membership implementation generic over storage?